src/v/cloud_storage/topic_mount_handler.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cloud_storage/remote.h"
+#include "cloud_storage/remote_label.h"
+#include "cloud_storage/remote_path_provider.h"
+#include "cloud_storage/topic_mount_manifest.h"
+#include "cluster/topic_configuration.h"
+#include "model/fundamental.h"
+#include "utils/retry_chain_node.h"
```

Comment:
nit: seems like most of these can be forward declared

Reply:
Fixed includes in `topic_mount_handler.h/.cc`.